pkg/codegen/go/gen.go

```diff
@@ -1618,7 +1618,9 @@ func (pkg *pkgContext) genResourceModule(w io.Writer) {
 	if topLevelModule {
 		fmt.Fprintf(w, "\tversion, err := PkgVersion()\n")
 	} else {
-		fmt.Fprintf(w, "\tversion, err := %s.PkgVersion()\n", pkg.pkg.Name)
+		// Some package names contain '-' characters, so grab the name from the base path.
+		pkgName := basePath[strings.LastIndex(basePath, "/")+1:]
```

Comment:
What do you expect this to make azure-nextgen and equinix-metal?

Reply:
I believe this is going off of the fact that our convention is: `github.com/pulumi/pulumi-azure-nextgen/sdk/go/azure` and `github.com/pulumi/pulumi-equinix-metal/sdk/go/equinix`. So the pkgName would be `azure` and `equinix` repectively.

Reply:
@lblackstone could we find a way to add a test for this? Go codegen is pretty sensitive and buggy so would like test coverage as we make these fixes.

Reply:
I can't think of a great way to test this synthetically. Perhaps we should add some assertions in the codegen code to validate assumptions about the schema (e.g., raise an error if importBasePath isn't set, or if the package name doesn't match)

Reply:
> I can't think of a great way to test this synthetically. Perhaps we should add some assertions in the codegen code to validate assumptions about the schema (e.g., raise an error if importBasePath isn't set, or if the package name doesn't match)

Yeah I am fine with an assertion.

Reply:
> Yeah I am fine with an assertion.

I'll add that as a follow-on once we've rolled out the importBasePath change to all the providers.